src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,305 @@
+
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroup.getCourse(), null);
+
+        return ResponseEntity.ok().body(tutorialGroup);
+    }
+
+    /**
+     * POST /courses/:courseId/tutorial-groups : creates a new tutorial group.
+     *
+     * @param courseId      the id of the course to which the tutorial group should be added
+     * @param tutorialGroup the tutorial group that should be created
+     * @return ResponseEntity with status 201 (Created) and in the body the new tutorial group
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> create(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) throws URISyntaxException {
+        log.debug("REST request to create TutorialGroup: {} in course: {}", tutorialGroup, courseId);
+        if (tutorialGroup.getId() != null) {
+            throw new BadRequestException("A new tutorial group cannot already have an ID");
+        }
+
+        var course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        tutorialGroup.setCourse(course);
+
+        trimStringFields(tutorialGroup);
+        checkTitleIsUnique(tutorialGroup);
+        TutorialGroup persistedTutorialGroup = tutorialGroupRepository.save(tutorialGroup);
+
+        return ResponseEntity.created(new URI("/api/tutorial-groups/" + persistedTutorialGroup.getId())).body(persistedTutorialGroup);
+    }
+
+    /**
+     * DELETE courses/:courseId/tutorial-groups/:tutorialGroupId : delete a tutorial group.
+     *
+     * @param courseId        the id of the course to which the tutorial groups belong to
+     * @param tutorialGroupId the id of the tutorial group to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> delete(@PathVariable Long courseId, @PathVariable Long tutorialGroupId) {
+        log.info("REST request to delete a TutorialGroup : {}", tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        tutorialGroupRepository.deleteById(tutorialGroupFromDatabase.getId());
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, tutorialGroupFromDatabase.getTitle())).build();
```

Comment:
Usually we return ResponseEntity.noContent() here.

Reply:
Done in 9eca40ec5100cab4110ea542c7bfc5de7a6f440e